components/bounty/funding-section/retract-or-withdraw-modal/controller.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+
+import { useTranslation } from "next-i18next";
+
+import { useAppState } from "contexts/app-state";
+import { toastError, toastSuccess } from "contexts/reducers/change-toaster";
+
+import { NetworkEvents, StandAloneEvents } from "interfaces/enums/events";
+import { fundingBenefactor } from "interfaces/issue-data";
+
+import useApi from "x-hooks/use-api";
+import useBepro from "x-hooks/use-bepro";
+import { useBounty } from "x-hooks/use-bounty";
+
+import RetractOrWithdrawModalView from "./view";
+
+interface RetractOrWithdrawModalControllerProps {
+  show?: boolean;
+  onCloseClick: () => void;
+  funding: fundingBenefactor;
+}
+
+export default function RetractOrWithdrawModalController({
+  show = false,
+  onCloseClick,
+  funding,
+}: RetractOrWithdrawModalControllerProps) {
+  const { t } = useTranslation(["common", "funding", "bounty"]);
+
+  const [isExecuting, setIsExecuting] = useState(false);
+
+  const { processEvent } = useApi();
+  const { getDatabaseBounty } = useBounty();
+  const { handleRetractFundBounty, handleWithdrawFundRewardBounty } =
+    useBepro();
+
+  const { dispatch, state } = useAppState();
+
+  const isBountyClosed = !!state.currentBounty?.data?.isClosed;
+  const tokenSymbol = state.currentBounty?.data?.transactionalToken?.symbol;
+  const rewardTokenSymbol = state.currentBounty?.data?.rewardToken?.symbol;
+  const retractOrWithdrawAmount = isBountyClosed
+    ? funding?.amount
+        ?.dividedBy(state.currentBounty?.data?.fundingAmount)
+        .multipliedBy(state.currentBounty?.data?.rewardAmount)
+        ?.toFixed()
+    : funding?.amount?.toFixed();
+
+  function handleRetractOrWithdraw() {
```

Comment:
this should be separated into two functions, and we should re-render the page because we need new data for the funding request - after calling processEvents

Reply:
Is it really necessary to do this? because this modal is called at different times (notFunded/closed-bounty)